source/views/building-hours/stateful-list.js

```diff
@@ -92,24 +117,51 @@ export class BuildingHoursView extends React.Component<Props, State> {
       buildings = defaultData.data
     }
     this.setState(() => ({
-      buildings: groupBuildings(buildings),
+      buildings: loadAllBuildings(buildings, this.props.favoriteBuildings),
       now: moment.tz(CENTRAL_TZ),
     }))
   }
 
+  onUpdate = () => {
+    this.fetchData()
+    this.props.onSaveFavorites(this.props.favoriteBuildings)
+  }
+
   render() {
     if (this.state.error) {
       return <NoticeView text={`Error: ${this.state.error.message}`} />
     }
-
     return (
       <BuildingHoursList
         buildings={this.state.buildings}
+        favoriteBuildings={this.props.favoriteBuildings}
         loading={this.state.loading}
         navigation={this.props.navigation}
         now={this.state.now}
         onRefresh={this.refresh}
+        onUpdate={this.onUpdate}
       />
     )
   }
 }
+
+function mapStateToProps(state: ReduxState) : ReduxStateProps {
+  if (!state.buildings) {
+    return {favoriteBuildings: []}
+  }
+
+  return {
+    favoriteBuildings: state.buildings.favoriteBuildings,
+  }
+}
+
+function mapDispatch(dispatch): ReduxDispatchProps {
```

Comment:
This one will need to move in to the Detail screen, because we'll be calling onSaveFavorites from there.